superset/assets/src/explore/reducers/exploreReducer.js

```diff
@@ -36,9 +36,31 @@ export default function exploreReducer(state = {}, action) {
       };
     },
     [actions.SET_DATASOURCE]() {
+      const newFormData = {...state.form_data};
+      if (action.datasource.type != state.datasource.type) {
```

Comment:
a comment explaining that this maps equivalent params between sqla and druid might be worthwhile

Reply:
@etr2460 there's no mapping of params merely enabling/disabling.